src/cirrus/build.py

```diff
@@ -45,6 +67,13 @@ def main():
         'venv',
         'bin',
         'virtualenv')
+
+    # remove existing virtual env if building clean
+    if opts.clean is True and os.path.exists(venv_path):
+        cmd = "rm -r {0}".format(venv_path)
```

Comment:
Maybe you are fine.  Didn't see the fabric stuff until now.  I'll let others comment on this.